Export libraries to trigger hooks.

nuclearsandwich opened the pull request:
Add `ament_export_libraries` calls to export libraries and set necessary environment hooks. The rviz packages are some of a handful which install libraries exporting them.

In conversation with @wjwwood we surmised that this was likely a small oversight rather than something that was done intentionally.

Connects to ros2/ros_workspace#10

nuclearsandwich committed:
Export libraries to trigger hooks.

wjwwood commented on rviz_default_plugins/CMakeLists.txt:
```diff
@@ -241,6 +241,7 @@ ament_target_dependencies(rviz_default_plugins
 
 ament_export_include_directories(include)
 ament_export_interfaces(rviz_default_plugins)
+ament_export_libraries(rviz_default_plugins)
```
I wonder if this is redundant with `ament_export_interfaces`, or rather whether or not that function should also extend the library paths.

@dirk-thomas what do you think?

dirk-thomas replied:
While both do kind of the same thing they work differently:
* `ament_export_interfaces` is used to export the targets via CMake `install(EXPORT ...)`
*  `ament_export_libraries` exports libraries using the variable `<pkgname>_LIBRARIES`

Both make sense depending on how downstream packages want to use them.

Since the target is a library `ament_export_interfaces` should pass the option `HAS_LIBRARY_TARGET` (see https://github.com/ament/ament_cmake/blob/88add8cab0ca579e89e95b3ab343af319c163195/ament_cmake_export_interfaces/cmake/ament_export_interfaces.cmake#L22-L24).

`HAS_LIBRARY_TARGET`

nuclearsandwich committed:
Use HAS_LIBRARY_TARGET with ament_export_interfaces.

Replaces ament_export_libraries with the HAS_LIBRARY_TARGET option for
ament_export_interfaces.

nuclearsandwich commented:
I've updated this PR to use the HAS_LIBRARY_TARGET option _rather_ than ament_export_libraries as all I needed was the environment hook to be set.

I swapped rather than did both since I think it's reasonable to make the minimal change unless maintainers prefer otherwise.

nuclearsandwich commented:
* Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=5275)](http://ci.ros2.org/job/ci_linux/5275/)
* Linux-aarch64 [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-aarch64&build=2031)](http://ci.ros2.org/job/ci_linux-aarch64/2031/)
* macOS [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_osx&build=4391)](http://ci.ros2.org/job/ci_osx/4391/)
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=5254)](http://ci.ros2.org/job/ci_windows/5254/)

wjwwood commented:
@mjcarroll you might want to consider this pr in the migration of the yaml cpp stuff.
